### ISSUE TITLE
Intranet webpages not working

### ISSUE THREAD

**Issue by beanonymous** [stestagg/dashcast#4]
Is there a way to cast intranet webpages via Dashcast?

Currently it currently does not work if the URL is not seen publicly online.

**Comment by stestagg** [comment]
It should work fine, as long as the chromecast is connected to the _same_ network as the intranet page.

If the chromecast is connected to the internet, then you're probably out-of-luck, as putting an intranet page onto the public internet is going to be a security issue for most people.

**Comment by beanonymous** [comment]
I figured out the issue. 

The dashcast or the Chromecast whichever, uses an online DNS and as a result can not find the internal network domain that our intranet uses. It works fine if I tell dashcast to connect to the internal IP but not via the internal network domain name.

Is it possible for future builds of dashcast to use the internal network DNS instead?

**Comment by mrothenbuecher** [comment]
i don't think so because the dns resolution depends on the chromecast not on the app which this app is running.

Use the ip adress of the chrome cast instead of the name

**Comment by XenonPK** [comment]
Chromecast always uses google dns, so if you have endpoints that only resolve locally, the chromecast won't be able to resolve them, because the upstream google dns servers do not know about your internal dns records.

This works fine if you have setup your firewall to redirect all dns traffic to your dns servers. I have tested this in my internal network and can confirm this is working.
The moment I remove the iptables rule it stops working, which is expected, considering the chromecast is querying a different dns server.

There were some early versions of the initial chromecast that were jailbroken (this was patched early on) and allowed you to have some control over the system configuration, but the most reliable option you have to enforce this is really using your firewall to ensure the traffic goes where you want.

This issue should probably be closed, since it is not really fixable by dashcast itself.
